pcsx2-qt/Debugger/DebuggerWindow.cpp

```diff
@@ -47,11 +47,12 @@ DebuggerWindow::DebuggerWindow(QWidget* parent)
 	m_ui.cpuTabs->addTab(m_cpuWidget_r3000, "R3000");
 
 	CBreakPoints::SetUpdateHandler(std::bind(&DebuggerWindow::onBreakpointsChanged, this));
-
-	return;
 }
 
-DebuggerWindow::~DebuggerWindow() = default;
+DebuggerWindow::~DebuggerWindow()
+{
+	CBreakPoints::SetUpdateHandler(nullptr);
```

Comment:
There's a race here - you're reading the update handler above on the CPU thread, but the window gets destructed on the UI thread.

It's a bit of a scary one too, because you could be calling into a freed object. The "best" way I can think of to handle this, would be to have the update handler queue a call back to the UI thread, which queries whether the debugger window is present, and if so, calls the method. Rather than the direct bind/call above.